tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -333,16 +333,44 @@ port::Status GetLoadedCudnnVersion(CudnnVersion* version) {
   return ::tensorflow::OkStatus();
 }
 
-#if CUDNN_MAJOR >= 8 && (CUDNN_MINOR > 0 || CUDNN_PATCHLEVEL >= 4)
-void PreloadCudnnLibrary(cudnnStatus_t (*version_check_fn)(),
-                         absl::string_view sub_library) {
-  cudnnStatus_t status = version_check_fn();
-  if (status != CUDNN_STATUS_SUCCESS) {
-    VLOG(1) << "Could not pre-initialize cuDNN sub-library " << sub_library
-            << ".  Error: " << cudnnGetErrorString(status) << ".";
+// Preload sub libs for cudnn 8.0.4+
+void PreloadCudnnSubLibs(const std::string& kind) {
+#if CUDNN_VERSION >= 8004
+  if (kind == "conv_fwd") {
+    cudnnOpsInferVersionCheck();
+    cudnnCnnInferVersionCheck();
+  } else if (kind == "conv_bwd_filter" || kind == "conv_bwd_data") {
```

Comment:
Nit: if you decided to use an enum you you can use a fall-through case and annotate it with `[[fallthrough]]`. See [gudie](https://google.github.io/styleguide/cppguide.html#Loops_and_Switch_Statements).